pkg/apiserver/resthandler.go

```diff
@@ -787,7 +787,15 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 		if admit != nil && admit.Handles(admission.Delete) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
+			var currentObject runtime.Object
```

Comment:
@liggitt I'll explore extending precondition checking on deletes, that does seem to be a better approach.